tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -128,6 +128,61 @@ def test__create_collection__InvalidParameters(self):
                     self.assertIn(error, response.json["detail"])
 
 
+class TestGetCollections(BaseAuthAPITest):
+    def setUp(self):
+        super().setUp()
+        self.test_collection = dict(
+            name="collection", description="description", contact_name="john doe", contact_email="johndoe@email.com"
+        )
+
+    def test__get_collections_no_auth__OK(self):
+        res_no_auth = self.app.get("/curation/v1/collections")
+        self.assertEqual(200, res_no_auth.status_code)
+        self.assertEqual(6, len(res_no_auth.json["collections"]))
+        [self.assertEqual("PUBLIC", c["visibility"]) for c in res_no_auth.json["collections"]]
+
+    def test__get_collections_no_auth_processing_status_PENDING__OK(self):
+        res_no_auth = self.app.get("/curation/v1/collections")
+        self.assertEqual(200, res_no_auth.status_code)
+        self.assertEqual(6, len(res_no_auth.json["collections"]))
+        for collection in res_no_auth.json["collections"]:
+            if collection["id"] == "test_collection_id":
+                self.assertEqual(collection["processing_status"], "PENDING")
+            else:
+                self.assertEqual(collection["processing_status"], "SUCCESS")
+        [self.assertEqual("PUBLIC", c["visibility"]) for c in res_no_auth.json["collections"]]
+
+    def test__get_collections_with_auth__OK(self):
+        res_auth = self.app.get("/curation/v1/collections", headers=self.get_auth_headers())
+        self.assertEqual(200, res_auth.status_code)
+        self.assertEqual(6, len(res_auth.json["collections"]))
```

Comment:
shouldn't this be 7, because we get both the public and private collections?

Reply:
No; originally I had it that way, but then I noticed I noticed [this](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/2083#:~:text=When%20the%20visibility%20parameter%20is%20unspecified%20or%20set%20to%20%22PUBLIC%22%2C%20a%20list%20of%20all%20public%20collections%20is%20returned.) from Brian

Reply:
waiting for confirmation from Brian on this